lib/datadistributor.js

```diff
@@ -54,10 +54,21 @@ define(['filters/nodefilter'], function (NodeFilter) {
     }
 
     function addFilter(d) {
-      filters.push(d);
-      notifyObservers();
-      d.setRefresh(refresh);
-      refresh();
+      var newItem = true;
+
+      filters.forEach(function (f) {
+        if (f.unique === d.unique) {
+          removeFilter(f);
```

Comment:
i thought about that, but if you have 2 filters it would be possible to end up with no node = no filter left.